src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageSupport.java

```diff
@@ -70,16 +73,39 @@ protected CompletableFuture<List<LSPUsagePsiElement>> doLoad(LSPUsageSupportPara
                         return CompletableFuture.completedFuture(null);
                     }
 
-                    DeclarationParams declarationParams = new DeclarationParams(textDocumentIdentifier, params.position());
-                    DefinitionParams definitionParams = new DefinitionParams(textDocumentIdentifier, params.position());
-                    TypeDefinitionParams typeDefinitionParams = new TypeDefinitionParams(textDocumentIdentifier, params.position());
-                    ReferenceParams referenceParams = createReferenceParams(textDocumentIdentifier, params.position(), project);
-                    ImplementationParams implementationParams = new ImplementationParams(textDocumentIdentifier, params.position());
+                    // Assume we need to collect everything
+                    boolean collectDefinitions = true;
+                    boolean collectTypeDefinitions = true;
+
+                    // If we can determine the type of element being searched, see if we can/should search for less
+                    Position position = params.position();
+                    Document document = LSPIJUtils.getDocument(file);
+                    int offset = document != null ? LSPIJUtils.toOffset(position, document) : -1;
+                    if (offset > -1) {
+                        LSPSemanticTokensFileViewProvider semanticTokensFileViewProvider = LSPSemanticTokensFileViewProvider.getInstance(file);
+                        if (semanticTokensFileViewProvider != null) {
+                            // Don't collect non-type definitions for type elements
+                            ThreeState isType = semanticTokensFileViewProvider.isType(offset);

```

Comment:
This is why we need `isType()` to be a `ThreeState`. We only want to filter if we _absolutely, positively_ know that the searched identifier is a type or is not a type based on LSP's predefined `SemanticTokenTypes` definitions.